lib/HTFeed/JobMetrics.pm

```diff
@@ -22,273 +28,178 @@ get_instance() method instead of a "normal" new() function.
 The singleton instance is stored in the aptly named
 $singleton variable.
 
+CLI provided by and documented in JobMetricsCLI.pm.
+
 =item See e.g.:
 
   https://www.perl.com/article/52/2013/12/11/Implementing-the-singleton-pattern-in-Perl/
 
 =back
 
-=item Command line options:
-
-=back
-
-=item --help / --usage
-
-Display the POD snippets in this file as a help document.
-
 =cut
 
-# If called from terminal:
-if (!caller) {
-    GetOptions(
-	# these are all booleans
-	'clear'    => \my $clear,
-	'help'     => \my $help,
-	'list'     => \my $list,
-	'location' => \my $location,
-	'pretty'   => \my $pretty,
-	'usage'    => \my $usage,
-	# Each operation takes a metric and/or a value:
-	# add(metric, value),
-	# get_value(metric)
-	# inc(metric),
-	# match(value),
-	# observe(metric, value)
-	'operation=s' => \my $operation,
-	'metric=s'    => \my $metric,
-	'value=s'     => \my $value
-    );
-
-    if ($help or $usage) {
-	pod2usage(2);
-    }
-
-    my $job_metrics = HTFeed::JobMetrics->get_instance();
-    my %valid_operations = (
-	add       => sub { $job_metrics->add($metric, $value) },
-	inc       => sub { $job_metrics->inc($metric) },
-	match     => sub {
-	    my $matches = $job_metrics->match($value);
-	    print join("\n", @$matches) . "\n";
-	},
-	get_value => sub {
-	    my $value = $job_metrics->get_value($metric);
-	    if (defined $value) {
-		print "$value\n";
-	    } else {
-		print "no value found\n";
-	    }
-	},
-	observe => sub { $job_metrics->observe($metric, $value) }
-    );
-
-    # Check the boolean args and act if true.
-    if ($clear) {
-	print "Clearing data...\n";
-	$job_metrics->clear;
-	print "Data cleared.\n";
-    }
-    if ($list) {
-	print "# List of metrics:\n";
-	print join("\n", @{$job_metrics->list_metrics}) . "\n";
-    }
-    if ($location) {
-	print "# HTFeed::JobMetrics data stored in:\n";
-	print $job_metrics->loc . "\n";
-    }
-    if ($pretty) {
-	print $job_metrics->pretty . "\n";
-    }
-
-    # Now check the operation arg
-    if (defined $operation) {
-	if (exists $valid_operations{$operation}) {
-	    print "running operation $operation ...\n";
-	    $valid_operations{$operation}();
-	} else {
-	    print "Invalid operation, exiting.\n";
-	    exit(1);
-	}
-    } else {
-	print "No operation defined, exiting.\n";
-	exit(0);
-    }
-}
-# end commandline stuff
-
 my $singleton = undef;
 
+# This class implements the singleton pattern.
+# Call get_instance instead of new.
```

Comment:
@moseshll had preferred `new` on the imgsrv side; we discussed a bit and he convinced me (https://github.com/hathitrust/babel/pull/13#discussion_r1664646616). Doesn't need to block merging, probably worth an issue to change it later.

Reply:
I don't care either way. I'm happy to change back to `new` for the sake of consistency.

Reply:
fixed in [de8df11](https://github.com/hathitrust/feed/pull/128/commits/de8df11fdd0fab74b8cbfb0632ea81c8d8aa0e7c)